File: main/algorithms/math/gcd/python/gcd.py
#Euclidian algorithm
#Finds greatest common division
def gcd(a,b):

    while a!=0 and b!=0:

        if a > b:
            a = a % b

        else:
            b = b % a
            
    return a + b

print(gcd(20, 30)) #10
print(gcd(15, 30)) #15
print(gcd(480, 564)) #12

File: main/algorithms/math/gcd/go/gcd.go
package main

import "fmt"

//Function that finds greates common division
func gcd(a, b int) int {

    //While b not equals b
    for b != 0 {

        //Assign b to a
        //and calculate new b
        a, b = b, a % b
    }

    //Return a
    return a
}

func main() {

    //First test
    fmt.Println(gcd(33, 77)) //11

    //Second test
    fmt.Println(gcd(49865, 69811)) //9973
}
